### DISCUSSION TITLE
code plugin: Possibility to use something fancier than a textarea to edit raw HTML?

### DISCUSSION THREAD

**Discussion by ThiefMaster** [tinymce/tinymce#9038] [Ideas]
I was thinking about using e.g. the monaco editor (vscode-like editor) when switching to code edit mode, but after having a look at the [plugin that provides code editing](https://github.com/tinymce/tinymce/blob/develop/modules/tinymce/src/plugins/code/main/ts/ui/Dialog.ts) 

Looking at https://www.tiny.cloud/docs/ui-components/dialogcomponents/#panelcomponents I don't see any mention of being able to define custom components either.

So I wonder if there is some way I just didn't find that would allow me to create a customized version of the code plugin that uses a different widget instead of the textarea.

**Comment by TheSpyder** [comment]
I thought we had documented all of our components, but it turns out we missed `customeditor`. I don't have good documentation to offer, sorry, but comparing it's definition to `textarea` should give you some starting points for how to configure it. The old style is easier to get running quickly.
https://github.com/tinymce/tinymce/blob/2d8b95517a43d1f1967a2f3898371865a2fa18e4/modules/bridge/src/main/ts/ephox/bridge/components/dialog/CustomEditor.ts
https://github.com/tinymce/tinymce/blob/f61d071878146aee69f57bdb712c1a6d0e8c8def/modules/bridge/src/main/ts/ephox/bridge/components/dialog/Textarea.ts

**Reply by ThiefMaster** [reply]
Is there some kind of registry where I have to register the component so it's available by name? Or can I pass the component class directly?
